connector/src/it/scala/org/apache/spark/sql/cassandra/execution/CassandraDirectJoinSpec.scala

```diff
@@ -622,6 +622,14 @@ class CassandraDirectJoinSpec extends SparkCassandraITFlatSpecBase with DefaultC
     left.join(right, left("id") === right("id"))
   }
 
+  it should "work with field extractor after join" in compareDirectOnDirectOff{ spark =>
```

Comment:
I would add just one more test with a doubly nested struct, I don't think it has to be full integration test like this, just something to show the getAlias method works on a very nested struct

Reply:
Added a doubly nested test.